src/__tests__/index.node.js

```diff
@@ -526,6 +527,80 @@ test('middleware - valid endpoint failure', async t => {
   t.end();
 });
 
+test('middleware - valid endpoint failure with standard error', async t => {
+  const mockCtx: Context = ({
+    headers: {},
+    prefix: '',
+    path: '/api/test',
+    method: 'POST',
+    body: {},
+    request: {
+      body: 'test-args',
+    },
+    memoized: new Map(),
+  }: any);
+  const e = new Error('Test Failure');
+  // $FlowFixMe
+  e.code = 'ERR_CODE_TEST';
+  // $FlowFixMe
+  e.meta = {hello: 'world'};
+  const mockHandlers = {
+    test() {
+      return Promise.reject(e);
+    },
+  };
+  const mockEmitter = {
+    emit(type, payload) {
+      t.equal(type, 'rpc:method');
+      t.equal(payload.method, 'test');
+      t.equal(payload.origin, 'browser');
+      t.equal(payload.status, 'failure');
+      t.equal(typeof payload.timing, 'number');
+      t.notEqual(payload.error, e);
+    },
+    from() {
+      return this;
+    },
+  };
+
+  const middleware =
+    RPCPlugin.middleware &&
+    RPCPlugin.middleware(
+      {
+        emitter: mockEmitter,
+        handlers: mockHandlers,
+      },
+      mockService
+    );
+  if (!middleware) {
+    t.fail();
+    t.end();
+    return;
+  }
+
+  try {
+    await middleware(mockCtx, () => Promise.resolve());
+    t.equal(
+      // $FlowFixMe
```

Comment:
Is there a benefit to flow testing our test files?

Reply:
can't hurt

Reply:
👍 

In many cases, our exports are not type checked until a downstream app depends upon it _except_ in their test files.  Although not perfect (i.e. loose type definitions may cause no issues, and still be incorrect in their specificity).  This also likely protects (albeit minimally) from type regressions.